src/Contract/Transaction.purs

```diff
@@ -253,31 +421,79 @@ derive newtype instance Eq BalancedSignedTransaction
 instance Show BalancedSignedTransaction where
   show = genericShow
 
+-- | Like 'balanceAndSignTx, but for more than one transaction.
+-- | This function may throw errors through the contract Monad.
+-- | If successful, transaction inputs will be locked afterwards.
+-- | If you want to re-use them in the same 'QueryM' context, call
+-- | 'unlockTransactionInputs'.
+balanceAndSignTxs
+  :: forall (r :: Row Type)
+   . Array UnattachedUnbalancedTx
+  -> Contract r (Array BalancedSignedTransaction)
+balanceAndSignTxs txs = do
+  txs' <- balanceTxs txs
+  let datumss = map (_.datums <<< unwrap) txs
+  traverse signOne (Array.zip txs' datumss)
+
+  where
+  signOne
+    :: Tuple UnattachedTransaction (Array Datum)
+    -> Contract r BalancedSignedTransaction
+  signOne (Tuple (balancedTx /\ redeemersTxIns) datums) = do
+    let inputs = balancedTx ^. _body <<< _inputs
+    redeemers <- liftedE $ reindexSpentScriptRedeemers inputs redeemersTxIns
+    -- Reattach datums and redeemer:
+    QueryM.FinalizedTransaction txCbor <-
+      liftedM "balanceAndSignTx: Cannot attach datums and redeemer"
+        (finalizeTx balancedTx datums redeemers)
+    -- Sign the transaction returned as Cbor-hex encoded:
+    signedTxCbor <- liftedM "balanceAndSignTx: Failed to sign transaction" $
+      signTransactionBytes (wrap txCbor)
+    pure $ BalancedSignedTransaction { transaction: balancedTx, signedTxCbor }
+
+-- | A helper that wraps a few steps into: balance an unbalanced transaction
+-- | (`balanceTx`), reindex script spend redeemers (not minting redeemers)
+-- | (`reindexSpentScriptRedeemers`), attach datums and redeemers to the
+-- | transaction (`finalizeTx`), and finally sign (`signTransactionBytes`).
+-- | The return type includes the balanced (but unsigned) transaction for
+-- | logging and more importantly, the `ByteArray` to be used with `Submit` to
+-- | submit the transaction.
+-- | If successful, transaction inputs will be locked afterwards.
+-- | If you want to re-use them in the same 'QueryM' context, call
+-- | 'unlockTransactionInputs'.
+balanceAndSignTxE
+  :: forall (r :: Row Type)
+   . UnattachedUnbalancedTx
+  -> Contract r (Either Error BalancedSignedTransaction)
+balanceAndSignTxE tx = try $ do
+  txs' <-
+    balanceAndSignTxs [ tx ] :: Contract r (Array BalancedSignedTransaction)
+  tx' <- headE txs'
+  pure tx'
+  where
+  headE :: forall a. Array a -> Contract r a
+  headE ar =
```

Comment:
Totally a stylistic thing, but if you want to avoid introducing a bound name for the array argument only to use it in a `case` expression, you can do 

```suggestion
  headE = case _ of
```

(and remove the following line of course)